### ISSUE TITLE
Animal Overcrowding and Drops Toggle

### ISSUE THREAD

**Issue by PsychoLynx** [MLG-Fortress/ExtraHardMode#159]
Hello folks of EHM!

I am not sure if this is a but or the mechanic working as intended, nervertheless I wanted to bring it up. First of all, I am a fan of using the Animal Overcrowind control setting to limit the amount of mobs (and give those poor chickens some breathing room!). 

However, players have been complaining that they no longer receive as many drops from their animals as before and I myself have also witnessed this. I was wondering if this has something to do with the Overcrowding feature and the damage it gives to animals in crowded places - it seems that way, since toggling the setting off seemed to make the drops normalize again. I suspect that animals who have recieved ANY damage from overcrowding don't drop their natural drops.

Tl;dr When Animal overcrowding is enabled, animals (who have probably been damaged by crowding) don't seem to drop as many things as they usually do, if at all. Is this intended behavior and if it is, can there be another toggle to allow overcrowding control and normal drops at the same time

**Comment by RoboMWM** [comment]
Can you test this?

**Comment by PsychoLynx** [comment]
I looked into it using the same test server as here (https://github.com/MLG-Fortress/ExtraHardMode/issues/163), only difference was adding a mob health bar plugin to see the chickens' health. 

My hypothesis was supported. After building a pen and spawning 16 chickens in there, waiting for some of them to get damaged by crowrding and then opening the pen the results were following:

A) All chickens with damage dropped nothing
B) All chickens without damage gave normal drops

**Comment by RoboMWM** [comment]
I'm guessing this is because the entities are tagged with environmental damage. Unsure if the rule is > 50% environmental damage = no drops or if it's any amount of damage.

Hmm, wonder if there is a reliable way to "untag" entities, since I don't think such a mechanism exists in the plugin. (And it effectively has added an environmental hazard with that feature, something it hasn't done before.)

**Comment by PsychoLynx** [comment]
In my testing the damage was often nowhere near 50%. Any damage from crowding seemed to cause no drops.

**Comment by PsychoLynx** [comment]
I tried this with the debug messages on, but nothing was printed.

**Comment by RoboMWM** [comment]
Debug messages only apply to mobgrinder atm.

**Comment by RoboMWM** [comment]
Can you test applying environmental damage to a mob, then observing if it later provides drops on player kill?

**Comment by PsychoLynx** [comment]
Thank you, works great now! Thank you so much!
